ci/datamgr.py

```diff
@@ -44,22 +128,182 @@ def get_logger(name, level=None, format=None, propagate=False):
 logger = get_logger(Path(__file__).with_suffix('').name)
 
 
-def recreate_database(driver, params, **kwargs):
-    database = params.pop("database", None)
-    url = sa.engine.url.URL(driver, **params)
+def make_ibis_client(env):
+    return ibis.impala.connect(
+        host=env.impala_host,
+        port=env.impala_port,
+        hdfs_client=ibis.impala.hdfs_connect(
+            host=env.nn_host,
+            port=env.webhdfs_port,
+            verify=env.auth_mechanism not in ('GSSAPI', 'LDAP'),
+            user=env.webhdfs_user,
+        ),
+        pool_size=multiprocessing.cpu_count(),
+    )
+
+
+def raise_if_cannot_write_to_hdfs(con, env):
+    test_path = os.path.join(env.test_data_dir, ibis.util.guid())
+    con.hdfs.put(test_path, BytesIO(ibis.util.guid().encode('UTF-8')))
+    con.hdfs.rm(test_path)
+
+
+def upload_ibis_test_data_to_hdfs(con, data_path, env):
+    hdfs = con.hdfs
+    if hdfs.exists(env.test_data_dir):
+        hdfs.rmdir(env.test_data_dir)
+    hdfs.put(env.test_data_dir, data_path)
+
+
+def create_test_database(con, env):
```

Comment:
Prefix these utility functions with `[_]impala_...`?

Reply:
Done. Also deduplicated the avro and parquet creation functions.